frontend/src/components/layout/SideNavBar/NavIcon.tsx

```diff
@@ -0,0 +1,49 @@
+import TooltipWrapper from 'components/common/TooltipWrapper';
+import { Roles } from 'constants/index';
+import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
+import _ from 'lodash';
+import { Nav } from 'react-bootstrap';
+import styled from 'styled-components';
+
+interface INavIconProps {
+  icon: React.ReactElement;
+  text: string;
+  showText: boolean;
+  onClick: () => void;
+  roles?: Roles[];
+}
+
+/**
+ * Component that creates a nav, with an icon, and optional text.
+ * @param {INavIconProps} param0
+ */
+export const NavIcon = ({ icon, text, showText, onClick, roles }: INavIconProps) => {
+  const { hasRole } = useKeycloakWrapper();
+  const displayIcon = _.find(roles, (role: Roles) => hasRole(role));
+
+  return !roles?.length || displayIcon ? (
+    <Nav.Item onClick={onClick} data-testid={`nav-tooltip-${text}`}>
```

Comment:
These links don't do anything yet, will update in psp-1843 to actually trigger navigation.